apps/grader-host/src/lib/receiveFromQueue.js

```diff
@@ -78,29 +133,19 @@ module.exports = function (sqs, queueUrl, receiveCallback, doneCallback) {
         }
       },
       async () => {
-        const timeout = parsedMessage.timeout || config.defaultTimeout;
-        // Add additional time to account for pulling the image, downloading/uploading files, etc.
-        const newTimeout = timeout + config.timeoutOverhead;
-        await sqs.send(
-          new ChangeMessageVisibilityCommand({
-            QueueUrl: queueUrl,
-            ReceiptHandle: receiptHandle,
-            VisibilityTimeout: newTimeout,
-          }),
-        );
+        heartbeatAbortController = await startHeartbeat(sqs, queueUrl, receiptHandle);
       },
       (callback) => {
-        receiveCallback(
-          parsedMessage,
-          (err) => {
+        receiveCallback(parsedMessage, (err) => {
```

Comment:
I changed from multiple callbacks to a single one that takes an optional error argument, which IMO is more idiomatic.

Reply:
Agreed.